### PR TITLE
Update session management to create new sessions after 6 hours of ina…

### PR DESCRIPTION
**Pull request by abbadminhas**
…ctivity

### CODE REVIEW

File: tests/test_supabase_logging.py [hunk 1]
```diff
@@ -36,8 +36,8 @@ def test_create_new_session_when_inactive(mock_supabase):
     # Mock SupabaseLogger instance
     logger = SupabaseLogger(supabase_client=mock_supabase)
 
-    # Mock inactive session data (last active > 15 minutes ago)
-    inactive_time = (datetime.now(timezone.utc) - timedelta(minutes=20)).isoformat()
```
Comment (lordmq) [inline tests/test_supabase_logging.py:40]: I think we need this. If someone travels between time zones how will we handle that?

Especially when someone goes from ahead to back like Pakistan to uk or Pakistan to Canada?

**Review by lordmq — CHANGES_REQUESTED** [review]
I think we need time zones. If we don't can you explain how we will handle when someone travels?

File: tests/test_supabase_logging.py (hunk 1 above)
Comment (abbadminhas) [reply tests/test_supabase_logging.py:40]: I have fixed this and reinstated time zones.

I also had to change date inconsistencies logic as follows because unit test was failing:

```
try:
    last_active = datetime.fromisoformat(session["last_active"])
except ValueError:
    last_active = datetime.strptime(session["last_active"], "%Y-%m-%dT%H:%M:%S.%f%z")
```

I'd like to see an example of last_active date to which this condition applies: 
`last_active = datetime.strptime(session["last_active"], "%Y-%m-%dT%H:%M:%S.%f%z")`

I also think we should have an explicit unit test that checks for this edge case. 

Can we discuss as part of this pull request review?

**Review by lordmq — APPROVED** [review]
